src/login/logind-button.c

```diff
@@ -120,71 +164,91 @@ static int button_install_check_event_source(Button *b) {
 }
 
 static int long_press_of_power_key_handler(sd_event_source *e, uint64_t usec, void *userdata) {
-        Manager *m = ASSERT_PTR(userdata);
+        Button *b = ASSERT_PTR(userdata);
+        Manager *m = b->manager;
 
         assert(e);
 
+        if (!m)
+                return 0;
```

Comment:
btw, why did you insert these `if (!m) return 0;` checks here all over the place? I don't see how a Button object could ever exist without the `.manager` field initialized. Hence this check can be dropped. Or what am I missing?

(Or better, turned into an assertion, i.e. use `Manager *m = ASSERT_PTR(b->manager);` when you decalre it)